rnn/multiply_node_gp.cxx

```diff
@@ -0,0 +1,123 @@
+#include <cmath>
+#include <limits>
+using std::numeric_limits;
+
+#include <vector>
+using std::vector;
+
+#include "common/log.hxx"
+#include "multiply_node_gp.hxx"
+
+MULTIPLY_Node_GP::MULTIPLY_Node_GP(int32_t _innovation_number, int32_t _layer_type, double _depth)
+    : MULTIPLY_Node(_innovation_number, _layer_type, _depth) {
+    node_type = MULTIPLY_NODE_GP;
+    Log::debug("created node: %d, layer type: %d, node type: MULTIPLY_NODE_GP\n", innovation_number, layer_type);
+}
+
+MULTIPLY_Node_GP::~MULTIPLY_Node_GP() {
+}
+
+void MULTIPLY_Node_GP::input_fired(int32_t time, double incoming_output) {
+    inputs_fired[time]++;
+
+    ordered_input[time].push_back(incoming_output);
+
+    if (inputs_fired[time] == 1) {
+        input_values[time] = incoming_output;
+    } else {
+        input_values[time] *= incoming_output;
+    }
+    if (inputs_fired[time] < total_inputs) {
+        return;
+    } else if (inputs_fired[time] > total_inputs) {
+        Log::fatal(
+            "ERROR: inputs_fired on RNN_Node %d at time %d is %d and total_inputs is %d\n", innovation_number, time,
+            inputs_fired[time], total_inputs
+        );
+        exit(1);
+    }
+
+    Log::debug("node %d - input value[%d]: %lf\n", innovation_number, time, input_values[time]);
+
+    output_values[time] = input_values[time] * bias;
+
+    if (ordered_input[time].size() != inputs_fired[time]) {
+        Log::fatal("ERROR: size of total_input is not the same as ordered_inputs\n");
+        Log::fatal("total: %d ordered: %d\n", total_inputs, ordered_input.size());
+        exit(1);
+    }
+
+    double total;
+    for (int i = 0; i < total_inputs; i++) {
+        total = 1.0;
+        for (int j = 0; j < total_inputs; j++) {
+            if (j != i) {
+                total *= ordered_input[time][j];
+            }
+        }
+        ordered_d_input[time].push_back(total * bias);
+    }
+
+#ifdef NAN_CHECKS
+    if (isnan(output_values[time]) || isinf(output_values[time])) {
```

Comment:
`isfinite`

Reply:
Previous nodes like rnn_node have the form used.  I might just leave it here in this case.

Reply:
it should probably be changed there too 💀

Reply:
alright, I made the change to multiply_node, multiply_node_gp, and rnn_node